mock/src/main.rs

```diff
@@ -0,0 +1,137 @@
+use std::{fs::File, io::Write, path::PathBuf, time::Instant};
+
+use clap::Parser;
+use crypto::{hash::rpo::RpoDigest as Digest, FieldElement, Word};
+use miden_objects::{
+    builder::DEFAULT_ACCOUNT_CODE,
+    mock::{Immutable, MockChain, OnChain},
+};
+use rand::SeedableRng;
+use rand_pcg::Pcg64;
+
+#[derive(Parser, Debug)]
+struct Args {
+    /// Value used to seed the PRNG.
+    #[arg(long)]
+    seed: Option<u64>,
+
+    /// Output file
+    output: PathBuf,
+}
+
+fn seed_to_word(seed: &str) -> Word {
+    let seed_bytes = hex::decode(seed).unwrap();
+    let data = unsafe { FieldElement::bytes_as_elements(&seed_bytes) }.unwrap();
+    let seed: Word = [data[0], data[1], data[2], data[3]];
+
+    seed
+}
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    env_logger::init();
+
+    let args = Args::parse();
+
+    let small_rng = match args.seed {
+        Some(seed) => Pcg64::seed_from_u64(seed),
+        None => Pcg64::from_entropy(),
+    };
+
+    println!(
+        "Generating mock data, results will be saved to {}",
+        args.output.to_str().expect("invalid file name")
+    );
+
+    let mut output = File::create(args.output)?;
+
+    // Block 1
+    // --------------------------------------------------------------------------------------------
+    let mut mock_chain = MockChain::new(small_rng);
+    let start = Instant::now();
+
+    let faucet = mock_chain.build_fungible_faucet_with_seed(
+        seed_to_word("3ef982ea3dca3f89179e1e86ef1c263896ef1e07e76325ce7c69825046bf75ec"),
+        OnChain::Yes,
+        DEFAULT_ACCOUNT_CODE,
+        Digest::default(),
+    );
+    println!("Fungible faucet created {} [took: {}s]", faucet, start.elapsed().as_secs());
+    mock_chain.seal_block();
+
+    // Block 2
+    // --------------------------------------------------------------------------------------------
```

Comment:
We should probably create a couple more blocks to the chain (could be done in a subsequent PR, but let's create an issue for this).

- Block 3: we add 3 notes to the chain:
  - Notes 1 and 2 originate from account 1 and have account 2 as their destination (specified via `tag`).
  - Note 3 also originates from account 1, but has a 3rd account as its destination. This third account would not be a part of the chain (but we'd have an ID for it already).
- Block 4: account 2 consumed one of the notes sent to 2.

This way, the current state of the chain would be:

1. There is a single fungible faucet.
2. There are 2 accounts with balances.
3. There are 3 notes, one of which has already been consumed.
  a. One of the notes can be consumed by an existing account - and we could write a test for this.
  b. Another note can be consumed by a new account - and we'd be able to test account creation with this.

Reply:
https://github.com/0xPolygonMiden/miden-base/issues/248